azurelinuxagent/common/protocol/wire.py

```diff
@@ -897,7 +897,7 @@ def _remove_stale_agent_manifest(self, family, incarnation):
         agent_manifest = os.path.join(conf.get_lib_dir(), fn)
 
         if os.path.exists(agent_manifest):
-            fileutil.rm_files([agent_manifest])
+            os.unlink(agent_manifest)
```

Comment:
I'm very confused by this. fileutil.rm_files() is:
```
def rm_files(*args):
    for paths in args:
        # find all possible file paths
        for path in glob.glob(paths):
            if os.path.isfile(path):
                os.remove(path)
```
Also, `os.unlink()` and `os.remove()` are identical on Linux.

I wonder if the actual bug was the way you passed a single filename to fileutil.rm_files; if I understand the Python correctly, you shouldn't have wrapped it in an array. I can imagine that glob.glob() behaves unexpectedly when passed an array.

In other words: is the correct fix
```
if os.path.exists(agent_manifest):
    fileutil.rm_files(agent_manifest)
```
?

Reply:
Yes, you are right.  This is the correct fix if I had kept rm_files.